### DISCUSSION TITLE
Wanted to understand logic behind DateTimeStamp constructor.

### DISCUSSION THREAD

**Discussion by anantk24** [microsoft/gctoolkit#42] [Q&A] [answered]
`public DateTimeStamp(ZonedDateTime dateTime, double timeStamp) {
        this.dateTime = dateTime;
        //NaN is our agreed upon not set but less than 0 makes no sense either.
        if ( dateTime != null && (Double.isNaN(timeStamp) || timeStamp < 0.0d))
            this.timeStamp = (double)dateTime.toEpochSecond() + (double) dateTime.getNano() / 1_000_000_000d;
}`

Why we have divided 1_000_000_000d in code ? Sorry, May be I missing some basics, I didn't understood.

**Comment by dsgrieve** [comment, marked as answer]
Converts the int nano value to a decimal nano value. The timeStamp value is in decimal seconds - `<seconds>.<millis>`

**Reply by karianna** [reply]
Sounds like a PR containing a helpful comment is in the making... ;-)

**Reply by anantk24** [reply]
@karianna, ok will add that in upcoming PR. 
@dsgrieve Thanks for the answer.
